Feature/cldsrv 244/migrate to GitHub actions

alexanderchan-scality opened the pull request:
Migrate CI from Eve to Github Actions:
* multiple-backend-tests
* file-ft-tests
* kmip-ft-tests
* utapi-v2-tests

tcarmet committed:
CLDSRV-115 migration github actions

tcarmet committed:
CLDSRV-115 migration to gh actions

bert-e commented:
# Hello alexanderchan-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/cloudserver/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-244 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.21`

* `7.70.12`

* `8.4.18`

* `8.5.19`

* `8.6.5`

* `8.7.2`


Please check the `Fix Version/s` of CLDSRV-244, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-244 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.21`

* `7.70.12`

* `8.4.18`

* `8.5.19`

* `8.6.6`

* `8.7.3`


Please check the `Fix Version/s` of CLDSRV-244, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-244 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.23`

* `7.70.14`

* `8.4.19`

* `8.5.20`

* `8.6.7`

* `8.7.5`


Please check the `Fix Version/s` of CLDSRV-244, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-244 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.24`

* `7.70.15`

* `8.4.19`

* `8.5.20`

* `8.6.7`

* `8.7.5`


Please check the `Fix Version/s` of CLDSRV-244, or the target
branch of this pull request.

alexanderchan-scality committed:
CLDSRV-244: migrate eve CI to GitHub Actions

- multibackend functional tests
- file functional tests
- utapi v2 functional tests
- kmip functional tests

alexanderchan-scality committed:
remove eve CI files

alexanderchan-scality committed:
remove duplications

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.4/feature/CLDSRV-244/migrateToGithubActions` with contents from `w/7.70/feature/CLDSRV-244/migrateToGithubActions`
and `development/8.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.4/feature/CLDSRV-244/migrateToGithubActions origin/development/8.4
 $ git merge origin/w/7.70/feature/CLDSRV-244/migrateToGithubActions
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.4/feature/CLDSRV-244/migrateToGithubActions
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/CLDSRV-244/migrateToGithubActions` into
`development/7.10`
* [w/7.70/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/7.70...w/7.70/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/7.70` 
* [w/8.4/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.4...w/8.4/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.4` 
* [w/8.5/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.5...w/8.5/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.5` 
* [w/8.6/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.6...w/8.6/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.6` 
* [w/8.7/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.7...w/8.7/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.7` 



The following branches will **NOT** be impacted:

* `development/7.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

alexanderchan-scality commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.4/feature/CLDSRV-244/migrateToGithubActions` with contents from `w/7.70/feature/CLDSRV-244/migrateToGithubActions`
and `development/8.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.4/feature/CLDSRV-244/migrateToGithubActions origin/development/8.4
 $ git merge origin/w/7.70/feature/CLDSRV-244/migrateToGithubActions
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.4/feature/CLDSRV-244/migrateToGithubActions
```

bert-e commented:
# History mismatch



Merge commit #8d5a502601a41860172ded578024a9da58c63d26 on the integration branch
`w/8.4/feature/CLDSRV-244/migrateToGithubActions` is merging a branch which is neither the current
branch `feature/CLDSRV-244/migrateToGithubActions` nor the development branch
`development/8.4`.

It is likely due to a rebase of the branch `feature/CLDSRV-244/migrateToGithubActions` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexanderchan-scality commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.4/feature/CLDSRV-244/migrateToGithubActions` with contents from `w/7.70/feature/CLDSRV-244/migrateToGithubActions`
and `development/8.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.4/feature/CLDSRV-244/migrateToGithubActions origin/development/8.4
 $ git merge origin/w/7.70/feature/CLDSRV-244/migrateToGithubActions
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.4/feature/CLDSRV-244/migrateToGithubActions
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.5/feature/CLDSRV-244/migrateToGithubActions` with contents from `w/8.4/feature/CLDSRV-244/migrateToGithubActions`
and `development/8.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.5/feature/CLDSRV-244/migrateToGithubActions origin/development/8.5
 $ git merge origin/w/8.4/feature/CLDSRV-244/migrateToGithubActions
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.5/feature/CLDSRV-244/migrateToGithubActions
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/CLDSRV-244/migrateToGithubActions` into
`development/7.10`
* [w/7.70/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/7.70...w/7.70/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/7.70` 
* [w/8.4/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.4...w/8.4/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.4` 
* [w/8.5/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.5...w/8.5/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.5` 
* [w/8.6/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.6...w/8.6/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.6` 
* [w/8.7/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.7...w/8.7/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.7` 



The following branches will **NOT** be impacted:

* `development/7.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

tcarmet commented on .github/workflows/tests.yaml:
```diff
@@ -0,0 +1,309 @@
+        source: artifacts
+      if: always()
+
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1.6.0
+      - name: Login to GitHub Registry
+        uses: docker/login-action@v1.10.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Login to Registry
+        uses: docker/login-action@v1
+        with:
+          registry: registry.scality.com
+          username: ${{ secrets.REGISTRY_LOGIN }}
+          password: ${{ secrets.REGISTRY_PASSWORD }}
+      - name: Build and push cloudserver image
+        uses: docker/build-push-action@v2.7.0
+        with:
+          push: true
+          context: .
+          tags: |
+            ghcr.io/${{ github.repository }}/cloudserver:${{ github.sha }}
+            registry.scality.com/cloudserver-dev/cloudserver:${{ github.sha }}
+          cache-from: type=gha,scope=cloudserver
+          cache-to: type=gha,mode=max,scope=cloudserver
```
Optional, you can use this reusable workflow if you want, does the same as this job. https://github.com/scality/workflows#docker-build

tcarmet commented on .github/workflows/tests.yaml:
```diff
@@ -0,0 +1,309 @@
+        user: ${{ secrets.ARTIFACTS_USER }}
+        password: ${{ secrets.ARTIFACTS_PASSWORD }}
+        source: /tmp/artifacts
+      if: always()
```
This is optional as well, and should be done in another PR given the time we have:

I believe you can optimize those jobs with a [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs), with something that looks like this:
```yaml
jobs:
  tests:
    strategy:
      matrix:
        tests:
        # more var/settings can be added as needed
        - name: multiple-backend
          env:
            CLOUDSERVER_IMAGE: ghcr.io/${{ github.repository }}/cloudserver:${{ github.sha }}
            S3BACKEND: mem
            S3_LOCATION_FILE: /usr/src/app/tests/locationConfig/locationConfigTests.json
            S3DATA: multiple
            JOB_NAME: ${{ github.job }}
        - name: kmip-ft-tests
        ...
        # add the settings for all tests
    name: ${{ matrix.tests.name }}
    env: ${{ matrix.tests.env }}
    steps:
    # Define the steps a single time and use the ${{ matrix.tests.<var> }} as needed for templating
    
```

tcarmet reviewed (approved):
Overall this LGTM, I have one required change that can be done in a separate PR and some ideas for you to play with in the future.

## Required changes

Stuff that are missing but can be done in another PR
- [ ] You need a release workflow, which seems to be missing here. You can pretty much copy paste the one from your neighboors like [backbeat](https://github.com/scality/backbeat/blob/development/8.6/.github/workflows/release.yaml). 

## Optimization

- [ ] I would split the `linting-coverage` in another workflow, just to keep files smaller and ease up reviews. 
- [ ] It also seems that `linting-coverage` does more than linting, which you may dedicate a job for the unit tests that are executed there.
- [ ] Create a reusable workflow for the end to end tests of cloudserver (might be overkill though).

alexanderchan-scality commented:
thanks @tcarmet.
The release workflow will be added for CLDSRV-329 in separate PR that is specific to migrating the 8.x branches to actions.

bert-e commented:
# History mismatch



Merge commit #561280a59d35f989e176f41010620cf03d8952c9 on the integration branch
`w/7.70/feature/CLDSRV-244/migrateToGithubActions` is merging a branch which is neither the current
branch `feature/CLDSRV-244/migrateToGithubActions` nor the development branch
`development/7.70`.

It is likely due to a rebase of the branch `feature/CLDSRV-244/migrateToGithubActions` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexanderchan-scality commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.4/feature/CLDSRV-244/migrateToGithubActions` with contents from `w/7.70/feature/CLDSRV-244/migrateToGithubActions`
and `development/8.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.4/feature/CLDSRV-244/migrateToGithubActions origin/development/8.4
 $ git merge origin/w/7.70/feature/CLDSRV-244/migrateToGithubActions
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.4/feature/CLDSRV-244/migrateToGithubActions
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.5/feature/CLDSRV-244/migrateToGithubActions` with contents from `w/8.4/feature/CLDSRV-244/migrateToGithubActions`
and `development/8.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.5/feature/CLDSRV-244/migrateToGithubActions origin/development/8.5
 $ git merge origin/w/8.4/feature/CLDSRV-244/migrateToGithubActions
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.5/feature/CLDSRV-244/migrateToGithubActions
```

tcarmet commented:
> thanks @tcarmet.
> The release workflow will be added for [CLDSRV-329](https://scality.atlassian.net/browse/CLDSRV-329) in separate PR that is specific to migrating the 8.x branches to actions.

Alright perfect 👌 

I'll let you double check if its specific to 8.x, as it seems that federation 7.10 is pulling images from cloudserver 7.10: https://github.com/scality/Federation/blob/development/7.10/group_vars/all#L138

UPDATE: I just understood, it already exists as a github actions workflow for 7.10, so indeed the work will be to just adapt the code for 8.x context

[CLDSRV-329]: https://scality.atlassian.net/browse/CLDSRV-329?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/CLDSRV-244/migrateToGithubActions` into
`development/7.10`
* [w/7.70/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/7.70...w/7.70/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/7.70` 
* [w/8.4/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.4...w/8.4/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.4` 
* [w/8.5/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.5...w/8.5/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.5` 
* [w/8.6/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.6...w/8.6/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.6` 
* [w/8.7/feature/CLDSRV-244/migrateToGithubActions](https://github.com/scality/cloudserver/compare/development/8.7...w/8.7/feature/CLDSRV-244/migrateToGithubActions) will be merged into `development/8.7` 



The following branches will **NOT** be impacted:

* `development/7.4`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

alexanderchan-scality commented:
/approve

alexanderchan-scality commented:
ping

alexanderchan-scality commented:
@tcarmet will need some assistance merging this and possibly also with the removal of the eve webhook.

alexanderchan-scality commented:
ping

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/7.70`

* :heavy_check_mark: `development/8.4`

* :heavy_check_mark: `development/8.5`

* :heavy_check_mark: `development/8.6`

* :heavy_check_mark: `development/8.7`



The following branches will **NOT** be impacted:

* `development/7.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/7.70`

* :heavy_check_mark: `development/8.4`

* :heavy_check_mark: `development/8.5`

* :heavy_check_mark: `development/8.6`

* :heavy_check_mark: `development/8.7`



The following branches have **NOT** changed:

* `development/7.4`



Please check the status of the associated issue CLDSRV-244.

Goodbye alexanderchan-scality.
